### DISCUSSION TITLE
How to purge used styles from custom CSS files?

### DISCUSSION THREAD

**Discussion by howagain** [tailwindlabs/tailwindcss#12197] [Help]
I'm using a library that contains our enterprise's CSS files, not ideal but required for us to use for some elements. It's 30kb+ of raw CSS for every variant of buttons and other basic elements, plus some base64 encoded images for backgrounds. My goal is to get that size down. So I'd love to at least remove all the unused styles from our `import 'enterprise-styling.css'` statements if we don't reference a class name. 

We're currently using Astro with the tailwindcss plugin. Are there suggestions for how to do this? I was looking at PurgeCSS but it seems tailwind has its own JIT way to remove unused styles now, so I'm wondering if we could leverage that. I could also contribute something but it needs to work for people not using tailwind as well. 

If nothing else, does anyone have a link to the source code for the CSS purging, was searching but couldn't find it.

**Comment by wongjn** [comment]
In Tailwind v3, there is no purging – Tailwind only generates the rule if the corresponding class name is found in any `content` file.

With regards to trimming down your enterprise's CSS, you could look try to `@import` the stylesheet into one of Tailwind's `@layer base/components/utilities` like:

```css
@import "path/to/enterprise-styling.css" layer(utilities);
```
(You may need to use `postcss-import` or similar before Tailwind runs depending on how Astro resolves CSS `@import` rules).

This lets Tailwind know about the class rules within the file and effectively gives you the "purging" behavior you describe.
